Utils/src/HiddenSectorProducer.cc

```diff
@@ -196,19 +236,159 @@ double HiddenSectorProducer::calculateMT2(const edm::Handle<edm::View<reco::GenM
   );
 }
 
+void HiddenSectorProducer::matchJetsCands(edm::Handle<edm::View<pat::Jet>>& h_jets, edm::Handle<edm::View<reco::Candidate>>& h_cands, std::vector<std::vector<CLorentzVector> >& cands_out, std::vector<std::vector<int> >& pdgids_out) const {
+
+	//loop over PF candidate collection once: check every jet in every jet collection
+	//only PF candidates found in a jet collection will be kept
+	cands_out.resize(h_jets->size());
+	pdgids_out.resize(h_jets->size());
+	for(unsigned c = 0; c < h_cands->size(); ++c){
+		const auto& candPtr = h_cands->ptrs()[c];
+		//if this cand is kept, it will be appended to cands_out
+		for(unsigned j = 0; j < h_jets->size(); ++j){
+			const auto& jet = h_jets->at(j);
+			const auto& daughterPtrs = jet.daughterPtrVector();
+			if(cands_out[j].size()==daughterPtrs.size()) continue;
+			bool keep = false;
+			for(const auto& candJetDau : jet.daughterPtrVector()){
```

Comment:
use the existing `daughterPtrs` variable here instead of calling `daughterPtrVector()` again (costly)